plugins/destination/bigquery/client/types.go

```diff
@@ -2,46 +2,208 @@ package client
+		arrow.FixedWidthTypes.Boolean):
+		return bigquery.BooleanFieldType
+	case typeOneOf(dataType,
+		arrow.PrimitiveTypes.Int8,
+		arrow.PrimitiveTypes.Int16,
+		arrow.PrimitiveTypes.Int32,
+		arrow.PrimitiveTypes.Int64,
+		arrow.PrimitiveTypes.Uint8,
+		arrow.PrimitiveTypes.Uint16,
+		arrow.PrimitiveTypes.Uint32,
+		arrow.PrimitiveTypes.Uint64):
+		return bigquery.IntegerFieldType
+	case typeOneOf(dataType,
+		arrow.PrimitiveTypes.Float32,
+		arrow.PrimitiveTypes.Float64):
+		return bigquery.FloatFieldType
+	case typeOneOf(dataType,
+		arrow.BinaryTypes.String,
+		arrow.BinaryTypes.LargeString):
+		return bigquery.StringFieldType
+	case typeOneOf(dataType,
+		arrow.BinaryTypes.Binary,
+		arrow.BinaryTypes.LargeBinary):
+		return bigquery.BytesFieldType
+	case typeOneOf(dataType,
+		arrow.FixedWidthTypes.Date32,
+		arrow.FixedWidthTypes.Date64):
+		return bigquery.DateFieldType
+	case typeOneOf(dataType,
+		arrow.FixedWidthTypes.Timestamp_s,
+		arrow.FixedWidthTypes.Timestamp_ms,
+		arrow.FixedWidthTypes.Timestamp_us):
+		return bigquery.TimestampFieldType
+	case typeOneOf(dataType,
+		arrow.FixedWidthTypes.Timestamp_ns):
+		return bigquery.RecordFieldType
+	case typeOneOf(dataType,
+		arrow.FixedWidthTypes.Time32s,
+		arrow.FixedWidthTypes.Time32ms,
+		arrow.FixedWidthTypes.Time64us,
+		arrow.FixedWidthTypes.Time64ns):
+		return bigquery.TimeFieldType
+	case typeOneOf(dataType,
+		arrow.FixedWidthTypes.Duration_s,
+		arrow.FixedWidthTypes.Duration_ms,
+		arrow.FixedWidthTypes.Duration_us,
+		arrow.FixedWidthTypes.Duration_ns):
+		// BigQuery does not support intervals with precisions higher than seconds,
+		// and in the case of seconds the max value is not large enough to contain the
+		// max Arrow duration, so we store durations as plain integers. Users will need
+		// to cast or transform to interval, if necessary.
+		return bigquery.IntegerFieldType
+	case typeOneOf(dataType,
+		arrow.FixedWidthTypes.MonthInterval):
+		return bigquery.RecordFieldType
+	case typeOneOf(dataType,
+		arrow.FixedWidthTypes.DayTimeInterval):
+		return bigquery.RecordFieldType
+	case typeOneOf(dataType,
+		arrow.FixedWidthTypes.MonthDayNanoInterval):
+		return bigquery.RecordFieldType
+	case typeOneOf(dataType,
+		types.ExtensionTypes.Inet,
+		types.ExtensionTypes.MAC,
+		types.ExtensionTypes.UUID):
+		return bigquery.StringFieldType
+	case typeOneOf(dataType,
+		types.ExtensionTypes.JSON):
+		return bigquery.JSONFieldType
 	default:
-		panic("unknown type")
+		panic("unsupported data type: " + dataType.String())
+	}
+}
+
+func (c *Client) DataTypeToBigQuerySchema(dataType arrow.DataType) bigquery.Schema {
+	switch {
+	case dataType.ID() == arrow.STRUCT:
+		v := dataType.(*arrow.StructType)
+		fields := make([]*bigquery.FieldSchema, len(v.Fields()))
+		for i, field := range v.Fields() {
+			fields[i] = c.ColumnToBigQuerySchema(schema.Column{
+				Name: field.Name,
+				Type: field.Type,
+			})
+		}
+		return fields
+	case arrow.IsListLike(dataType.ID()):
```

Comment:
you should probably add a case for map handling above, as `arrow.IsListLike` matches maps, too

Reply:
We don't have tests for maps yet, so I haven't handled them in this PR